extract.R

```diff
@@ -0,0 +1,173 @@
+semtext <- function(fit, x2=T, cfi=F, tli=T, rmsea=T, rmsea.ci=T, srmr=T, bic=F, aic=F,
+                    clipboard=T, type = "scaled", srmr.type = "srmr", output="text",
+                    modelnames=NULL,
+                    sep = "\t", dec=",") {
+  ## fitted model or list of fitted model
+  ## type: standard , scaled , robust
+  ## srmr.type: srmr, srmr_bentler, srmr_bentler_nomean, srmr_bollen, srmr_bollen_nomean, srmr_mplus, srmr_mplus_nomean
+  
+  if(!is.null(modelnames)) {
+    model.names <- modelnames
+  } else {
+    model.names <- as.list(sys.call())[[2]]
+    model.names <- gsub("list", "", model.names)[-1]
+  }
+  
+  models <- length(fit)
+  fitvalue <- list()
+  
+  ## extract fitmeasuers
+  if (models == 1) {
+    fitvalue[[1]] <- fitmeasures(fit)
+  } else {
+    for (i in c(1:length(fit))) {
+      fitvalue[[i]] <- fitmeasures(fit[[i]])
+    }
+  }
+
+  
+  ## type of robust chi2
+  if (type == "robust") {
+    if(x2 == F) {x2 <- NULL} else {x2 <- c("chisq.scaled", "df.scaled", "pvalue.scaled")}
+    if(cfi == F) {cfi <- NULL} else {cfi <- "cfi.robust"}
+    if(tli == F) {tli <- NULL} else {tli <- "tli.robust"}
+    if(rmsea == F) {rmsea <- NULL} else {rmsea <- "rmsea.robust"}
+    if(rmsea.ci == F) {rmsea.ci <- NULL} else {rmsea.ci <- c("rmsea.ci.lower.robust", "rmsea.ci.upper.robust")}
+    if(srmr == F) {
+      srmr <- NULL
+    } else if (is.null(srmr.type)) {
+      srmr <- srmr
+    } else {srmr <- srmr.type}
```

Comment:
Co takhle se uplne vykaslat na to mit dva argumenty? Staci Ti jeden. Bud bude `srmr` obsahovat nektery z povolenych types, nebo ho neoutputnes (`if (srmr %in% c(...)) {} else {srmr <- NULL}`)

Reply:
Nechal bych srmr.type, ale srmr jako takové definoval skrze ten string "indices", jak to navrhuješ níže.